Fix: Enforce consistent, responsive `<figure>` styling to match markdown image behavior

zihanKuang opened the pull request:
**Notes for Reviewers**

This PR fixes #522

- [x] Yes, I signed my commits.

### Problem
The current `<figure>` styling does not adapt well when inline style="width:..." is present. In such cases, the figure can exceed the intended content width and introduce excessive horizontal gaps, especially on smaller screens.

### Resolution
This PR updates the styling of the `<figure>` element in `assets/scss/_styles_project.scss` to:

* Align with the behavior of markdown images rendered via `layouts/_default/_markup/render-image.html`
* Ensure consistent responsiveness by applying:

  * `width: 100%` and `max-width: 70%` to `figure` and `figure[style*="width"]`
  * `width: 100%` to nested `<img>` and `<figcaption>`
  * `object-fit: contain`, `display: block`, and centered alignment
* Override any inline width declarations (e.g., `style="width:600px"`) using `!important`

Although `<figure>` is infrequently used (standard markdown image syntax is more common), it remains useful for including captions or grouped content. To support this usage, this PR also updates **contributing-to-docs** with usage guidance and examples.

![image](https://github.com/user-attachments/assets/363e7fdd-a9e2-461a-913d-041c3b7f9031)

leecalcote commented:
Here is a comparison of before and after.

before:


https://github.com/user-attachments/assets/b3c164cb-1866-4492-ab58-682e07734f90


After:


https://github.com/user-attachments/assets/ea7d83be-24a5-403f-9b2f-481243b6f12d

I’m not positive that the after it is better in Mobile view… What do you think? The 100% with the mobile view might be a little more friendly.

M-DEV-1 commented:
I agree with Lee, the images are too small to be readable, 100% width would be nice.

M-DEV-1 commented on assets/scss/_styles_project.scss:
```diff
@@ -456,26 +456,38 @@ a:not([href]):not([class]):hover {
+figure,
+figure[style*="width"] {
+  width: 100% !important;
+  max-width: 70% !important;
```
LGTM, but do we really need both as `!important`?

M-DEV-1 commented on content/en/kanvas/reference/contributing-to-docs.md:
````diff
@@ -180,6 +180,17 @@ This default styling works well for most landscape (horizontal) images. However,
 <img src="./images/example.png" alt="Example description" style="max-width: 40vw; max-height: 60vh; display: block; margin: 1rem auto;" />
 ```
 
+If you want your image to include a caption for explanation or accessibility, you can use the `<figure>` element:
````
these instructions are good 👍🏼

M-DEV-1 commented on assets/scss/_styles_project.scss:
```diff
@@ -456,26 +456,38 @@ a:not([href]):not([class]):hover {
-    margin-bottom: 0px !important;
+    width: 100% !important;
+    height: auto !important;
+    object-fit: contain !important;
```
is `!important` on everything needed here?

zihanKuang commented:
Hi @leecalcote @M-DEV-1 

Thanks for the feedback!

1. The original looked better mainly because the review page used an inline width of 400px, which happened to be close to the screen size, so it looked like 100%. But on the performance page, the inline width is 600px, which made the image too large and not responsive.

2. I chose 70% because that's the width we use for rendering markdown images. I wanted the images across the page to be visually consistent and aligned.

3. 100% works better on mobile, while 70% looks better on desktop. So I added a media query to handle that, and also updated the markdown image rendering to match. Now both markdown and figure images are 100% width on screens smaller than 600px, and 70% on larger screens.

4. As for the 600px breakpoint, I checked the Meshery and Layer5 docs. There’s a wide range of data used as reference, and I picked 600px as a reasonable cutoff based on those examples.

Let me know if further adjustments are needed😎

M-DEV-1 commented:
Thank you @leecalcote,

@zihanKuang thanks for justifying the usage 👌
